src/Microsoft.AspNet.Mvc.ModelBinding/ModelStateDictionary.cs

```diff
@@ -135,7 +135,10 @@ public ICollection<ModelState> Values
         /// <inheritdoc />
         public bool IsValid
         {
-            get { return ValidationState == ModelValidationState.Valid; }
+            get
+            {
+                return ErrorCount == 0;
+            }
```

Comment:
I understand that, but the problem is this, A model is marked as Unvalidated even if any of its properties are unvalidated which means  if a model is skipped from validation, the state for all the properties of that model should be Skipped(assuming such a state existed) instead of Unvalidated, note that the model itself is still validated which means errors at the model level will still be valid/invalid/unvalidated ( they cannot be skipped). 
consider this:
validation is an atomic operation, moreover, model binding and validation are guaranteed to run for action parameters. For scenarios involving TryUpdateModel this is broken for now, the users will have to call TryValidateModel before checking the is valid error state, which is ok if we are going with statement that model binding and validation are two distinct process ( we can always add a method which does both for the user).